tests/gumjs/script.c

```diff
@@ -7083,6 +7090,86 @@ TESTCASE (interceptor_should_refuse_to_attach_without_any_callbacks)
       "Error: expected at least one callback");
 }
 
+#ifdef HAVE_DARWIN
+
+TESTCASE (interceptor_and_js_should_not_deadlock)
+{
+  GThread * worker_thread;
+  int state = 0;
+
+  if (!g_test_slow ())
+  {
+    g_print ("<skipping, run in slow mode> ");
+    return;
+  }
+
+  worker_thread = g_thread_new ("script-test-worker-thread",
+      interceptor_attacher_worker, &state);
+  while (state == 0)
+    g_usleep (G_USEC_PER_SEC / 200);
+
+  COMPILE_AND_LOAD_SCRIPT (
+      "const iterations = 100;"
+      "send('Start loop');"
+      "const threadSuspend = new NativeFunction("
+      "  Module.getExportByName(null, 'thread_suspend'),"
+      "  'int', ['int'], { scheduling: 'exclusive' }"
+      ");"
+      "Interceptor.replace(threadSuspend, new NativeCallback((threadId) => {"
+      "  return threadSuspend(threadId);"
+      "}, 'int', ['int']));"
+      "Interceptor.flush();"
+      "setTimeout(() => {"
+      "  for (let i = 0; i !== iterations; i++) {"
+      "    Thread.sleep(0.1);"
+      "  }"
+      "  Interceptor.revert(threadSuspend);"
+      "  send('The end');"
+      "}, 0);");
+
+  EXPECT_SEND_MESSAGE_WITH ("\"Start loop\"");
+
+  g_usleep (G_USEC_PER_SEC / 25);
+  g_thread_join (worker_thread);
+  g_assert_cmpint (state, ==, 2);
+  EXPECT_SEND_MESSAGE_WITH ("\"The end\"");
+  EXPECT_NO_MESSAGES ();
+}
+
+static gpointer
+interceptor_attacher_worker (gpointer data)
+{
+  int * state = data;
+  guint i;
+  GumInterceptor * interceptor;
+  TestCallbackListener * listener;
+  GumAttachReturn result;
+
+  *state = 1;
+
+  interceptor = gum_interceptor_obtain ();
+  listener = test_callback_listener_new ();
+
+  for (i = 0; i != 300; i++)
+  {
+    result = gum_interceptor_attach (interceptor, target_function_int,
```

Comment:
Would be good to use `gum_make_call_listener ()` here if possible.

Reply:
yes! way better!! thanks